calico/felix/endpoint.py

```diff
@@ -226,6 +233,15 @@ def on_endpoint_update(self, endpoint):
         if endpoint != self.endpoint:
             self._dirty = True
 
+        # When an IP address is removed, squash any old conntrack entries for
+        # that IP address.  This prevents the table from becoming full under
+        # high VM churn and it also prevents a new endpoint that happens to
+        # get a reused IP from spoofing old streams.
+        old_endpoint = self.endpoint or {}
+        old_ip_nets = set(old_endpoint.get(self.nets_key, []))
+        removed_nets = old_ip_nets - new_ip_nets
+        remove_conntrack_flows([futils.net_to_ip(n) for n in removed_nets])
+
```

Comment:
You are removing the conntrack entries while the interface still has them in place. I think it would be better to move this call into devices.set_routes, so you delete the conntrack entry at the same point as you delete the IP address itself.